kernel/poll.c

```diff
@@ -188,36 +188,77 @@ static inline void set_event_ready(struct k_poll_event *event, u32_t state)
 	event->state |= state;
 }
 
-int z_impl_k_poll(struct k_poll_event *events, int num_events, s32_t timeout)
+static inline int register_events(struct k_poll_event *events,
+				  int num_events,
+				  struct _poller *poller,
+				  bool just_check)
 {
-	__ASSERT(!z_is_in_isr(), "");
-	__ASSERT(events != NULL, "NULL events\n");
-	__ASSERT(num_events > 0, "zero events\n");
+	int events_registered = 0;
 
-	int last_registered = -1, rc;
-	k_spinlock_key_t key;
-
-	struct _poller poller = { .thread = _current, .is_polling = true, };
-
-	/* find events whose condition is already fulfilled */
-	for (int ii = 0; ii < num_events; ii++) {
+	for (int i = 0; i < num_events; i++) {
+		k_spinlock_key_t key;
 		u32_t state;
 
 		key = k_spin_lock(&lock);
-		if (is_condition_met(&events[ii], &state)) {
-			set_event_ready(&events[ii], state);
```

Comment:
Again lots of green and red all due to changes in variable names, obscuring the important differences.

Reply:
Fixed